# Library

In [106]:
import pandas as pd 
import plotly.express as px
import plotly.io as pio
from pandas_profiling import ProfileReport

C:\Users\WINDOWS\AppData\Local\Temp\ipykernel_28476\2783550639.py:4: DeprecationWarning:

`import pandas_profiling` is going to be deprecated by April 1st. Please use `import ydata_profiling` instead.



In [6]:
df = pd.read_csv('C:\\Users\\WINDOWS\\Downloads\\customer_booking.csv', encoding='latin1')

# AED

In [8]:
df.columns

Index(['num_passengers', 'sales_channel', 'trip_type', 'purchase_lead',
       'length_of_stay', 'flight_hour', 'flight_day', 'route',
       'booking_origin', 'wants_extra_baggage', 'wants_preferred_seat',
       'wants_in_flight_meals', 'flight_duration', 'booking_complete'],
      dtype='object')

In [9]:
df.dtypes

num_passengers             int64
sales_channel             object
trip_type                 object
purchase_lead              int64
length_of_stay             int64
flight_hour                int64
flight_day                object
route                     object
booking_origin            object
wants_extra_baggage        int64
wants_preferred_seat       int64
wants_in_flight_meals      int64
flight_duration          float64
booking_complete           int64
dtype: object

In [10]:
df.shape

(50000, 14)

## Missing_values

In [11]:
df.isnull().sum()

num_passengers           0
sales_channel            0
trip_type                0
purchase_lead            0
length_of_stay           0
flight_hour              0
flight_day               0
route                    0
booking_origin           0
wants_extra_baggage      0
wants_preferred_seat     0
wants_in_flight_meals    0
flight_duration          0
booking_complete         0
dtype: int64

## Duplicates_values

In [12]:
df.duplicated().sum()

719

In [13]:
df = df.drop_duplicates()

In [14]:
df.shape

(49281, 14)

In [16]:
df.dtypes

num_passengers             int64
sales_channel             object
trip_type                 object
purchase_lead              int64
length_of_stay             int64
flight_hour                int64
flight_day                object
route                     object
booking_origin            object
wants_extra_baggage        int64
wants_preferred_seat       int64
wants_in_flight_meals      int64
flight_duration          float64
booking_complete           int64
dtype: object

In [19]:
df.flight_duration.describe()

count    49281.000000
mean         7.279974
std          1.496390
min          4.670000
25%          5.620000
50%          7.570000
75%          8.830000
max          9.500000
Name: flight_duration, dtype: float64

In [18]:
df.length_of_stay.describe()

count    49281.000000
mean        23.053976
std         33.832149
min          0.000000
25%          5.000000
50%         17.000000
75%         28.000000
max        778.000000
Name: length_of_stay, dtype: float64

In [22]:
df.booking_origin.unique()

array(['New Zealand', 'India', 'United Kingdom', 'China', 'South Korea',
       'Japan', 'Malaysia', 'Singapore', 'Switzerland', 'Germany',
       'Indonesia', 'Czech Republic', 'Vietnam', 'Thailand', 'Spain',
       'Romania', 'Ireland', 'Italy', 'Slovakia', 'United Arab Emirates',
       'Tonga', 'Réunion', '(not set)', 'Saudi Arabia', 'Netherlands',
       'Qatar', 'Hong Kong', 'Philippines', 'Sri Lanka', 'France',
       'Croatia', 'United States', 'Laos', 'Hungary', 'Portugal',
       'Cyprus', 'Australia', 'Cambodia', 'Poland', 'Belgium', 'Oman',
       'Bangladesh', 'Kazakhstan', 'Brazil', 'Turkey', 'Kenya', 'Taiwan',
       'Brunei', 'Chile', 'Bulgaria', 'Ukraine', 'Denmark', 'Colombia',
       'Iran', 'Bahrain', 'Solomon Islands', 'Slovenia', 'Mauritius',
       'Nepal', 'Russia', 'Kuwait', 'Mexico', 'Sweden', 'Austria',
       'Lebanon', 'Jordan', 'Greece', 'Mongolia', 'Canada', 'Tanzania',
       'Peru', 'Timor-Leste', 'Argentina', 'New Caledonia', 'Macau',
       'Myanmar (

In [74]:
# Grouper par origine et statut de réservation
reservations_status = df.groupby(['booking_origin', 'booking_complete'])['booking_complete'].count()

# Réorganiser les résultats sous forme de tableau clair (optionnel)
reservations_status = reservations_status.unstack(fill_value=0)
reservations_status

booking_complete          0    1
booking_origin                  
(not set)                64   14
Afghanistan               1    0
Algeria                   1    0
Argentina                 6    0
Australia             16796  895
...                     ...  ...
United Arab Emirates     39    3
United Kingdom          152   21
United States           370   83
Vanuatu                   0    1
Vietnam                 272  114

[104 rows x 2 columns]

In [27]:
# Filtrer uniquement les réservations complètes
df_complete = df[df['booking_complete'] == 1]

# Grouper par origine et compter le nombre de réservations complètes
reservations_by_origin = df_complete.groupby('booking_origin')['booking_complete'].count()

# Trier les résultats par ordre décroissant pour voir les origines les plus actives
reservations_by_origin = reservations_by_origin.sort_values(ascending=False)

reservations_by_origin


booking_origin
Malaysia                2434
Australia                895
China                    684
Indonesia                609
Japan                    472
Thailand                 464
South Korea              459
Singapore                295
Taiwan                   220
India                    128
Vietnam                  114
Macau                     95
United States             83
Philippines               75
Hong Kong                 72
New Zealand               53
Brunei                    40
Cambodia                  36
United Kingdom            21
(not set)                 14
Myanmar (Burma)           13
France                    13
Germany                   11
Saudi Arabia              11
Italy                     11
Laos                       6
Sweden                     5
Qatar                      5
Mauritius                  4
Switzerland                3
Nepal                      3
Russia                     3
Bahrain                    3
Oman                       3

In [29]:
# Filtrer uniquement les réservations complètes
df_Incomplete = df[df['booking_complete'] == 0]

# Grouper par origine et compter le nombre de réservations complètes
reservations_by_origine = df_Incomplete.groupby('booking_origin')['booking_complete'].count()

# Trier les résultats par ordre décroissant pour voir les origines les plus actives
reservations_by_origine = reservations_by_origine.sort_values(ascending=False)

reservations_by_origine


booking_origin
Australia           16796
Malaysia             4621
South Korea          4043
Japan                3347
China                2600
                    ...  
Norfolk Island          1
Gibraltar               1
Panama                  1
Papua New Guinea        1
Bulgaria                1
Name: booking_complete, Length: 100, dtype: int64

In [49]:
reservationc_par_jour = df_complete.groupby('flight_day')['booking_complete'].count().reset_index()
reservationc_par_jour

flight_day  booking_complete
0        Fri               976
1        Mon              1194
2        Sat               855
3        Sun               911
4        Thu              1109
5        Tue              1116
6        Wed              1230

In [52]:
fig = px.bar(reservationc_par_jour, x='flight_day', y='booking_complete', 
             title='Réservations complètes par jour de vol',
             labels={'flight_day': 'Jour de vol', 'booking_complete': 'Nombre de réservations complètes'},
             color='flight_day', color_discrete_sequence=px.colors.qualitative.Set1)

# Afficher le graphique
fig.show()


In [61]:
reservationcIN_par_jour = df_Incomplete.groupby('flight_day')['booking_complete'].count().reset_index()
reservationcIN_par_jour

flight_day  booking_complete
0        Fri              5709
1        Mon              6794
2        Sat              4868
3        Sun              5531
4        Thu              6214
5        Tue              6442
6        Wed              6332

In [63]:
fig = px.bar(reservationcIN_par_jour, x='flight_day', y='booking_complete',
             title='Réservations incomplètes par jour de vol',
             labels={'flight_day': 'Jour de vol', 'booking_incomplete': 'Nombre de réservations incomplètes'},
             color='flight_day', color_discrete_sequence=px.colors.qualitative.Set1)

# Afficher le graphique
fig.show()

In [67]:
reservationc_par_heure_de_vol = df_complete.groupby('flight_hour')['booking_complete'].count().reset_index()
reservationc_par_heure_de_vol

flight_hour  booking_complete
0             0               180
1             1               284
2             2               386
3             3               384
4             4               419
5             5               429
6             6               482
7             7               422
8             8               478
9             9               507
10           10               410
11           11               401
12           12               451
13           13               499
14           14               485
15           15               371
16           16               293
17           17               162
18           18                84
19           19                39
20           20                25
21           21                45
22           22                52
23           23               103

In [ ]:
df.fli

In [72]:
fig = px.bar(reservationc_par_heure_de_vol, x='flight_hour',y='booking_complete')
fig.show()

In [70]:
reservationcIN_par_heure_de_vol = df_Incomplete.groupby('flight_hour')['booking_complete'].count().reset_index()
reservationcIN_par_heure_de_vol

flight_hour  booking_complete
0             0              1321
1             1              1787
2             2              2210
3             3              2232
4             4              2367
5             5              2388
6             6              2528
7             7              2658
8             8              2647
9             9              2590
10           10              2640
11           11              2670
12           12              2663
13           13              2547
14           14              2298
15           15              1830
16           16              1243
17           17               686
18           18               346
19           19               255
20           20               250
21           21               341
22           22               521
23           23               872

In [75]:
fig = px.bar(reservationcIN_par_heure_de_vol, x='flight_hour',y='booking_complete')
fig.show()

In [81]:
df['departure'] = df['route'].str[:3]
df['destination'] = df['route'].str[-3:]
# Placer 'departure' à l'indice 8
df.insert(8, 'departure', df.pop('departure'))

# Placer 'destination' à l'indice 9
df.insert(9, 'destination', df.pop('destination'))

C:\Users\WINDOWS\AppData\Local\Temp\ipykernel_28476\3017026656.py:1: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy

C:\Users\WINDOWS\AppData\Local\Temp\ipykernel_28476\3017026656.py:2: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [85]:
df.drop(columns='route',inplace=True)

C:\Users\WINDOWS\AppData\Local\Temp\ipykernel_28476\3359113153.py:1: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [94]:
passengers_by_reservation_complete = df_complete.groupby('num_passengers')['booking_complete'].count().reset_index()

In [95]:
fig = px.bar(passengers_by_reservation_complete, x='num_passengers',y='booking_complete')
fig.show()


In [97]:
passengers_by_reservation_incomplete = df_Incomplete.groupby('num_passengers')['booking_complete'].count().reset_index()

In [99]:
fig = px.bar(passengers_by_reservation_incomplete, x='num_passengers',y='booking_complete')
fig.show()

In [100]:
passengers_by_reservation_complete

num_passengers  booking_complete
0               1              4403
1               2              2029
2               3               486
3               4               273
4               5               106
5               6                53
6               7                16
7               8                12
8               9                13

In [101]:
passengers_by_reservation_incomplete

num_passengers  booking_complete
0               1             26476
1               2             10640
2               3              2396
3               4              1494
4               5               438
5               6               228
6               7                91
7               8                76
8               9                51

# sales_channel

In [129]:
chanel_by_complete =  df_complete.groupby('sales_channel')['booking_complete'].count().reset_index()
chanel_by_complete

sales_channel  booking_complete
0      Internet              6802
1        Mobile               589

In [124]:
chanel_by_incomplete =  df_Incomplete.groupby('sales_channel')['booking_complete'].count().reset_index()
chanel_by_incomplete

sales_channel  booking_complete
0      Internet             37115
1        Mobile              4775

In [107]:
# Créer le rapport pour df_Incomplete
profile_incomplete = ProfileReport(df_Incomplete, title="Profiling Report: Incomplete Bookings")

# Créer le rapport pour df_complete
profile_complete = ProfileReport(df_complete, title="Profiling Report: Complete Bookings")

# Comparer les deux rapports
comparison_report = profile_incomplete.compare(profile_complete)

# Exporter le rapport comparatif en fichier HTML
comparison_report.to_file("comparison.html")

c:\Users\WINDOWS\AppData\Local\Programs\Python\Python312\Lib\site-packages\ydata_profiling\utils\dataframe.py:137: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



Summarize dataset:   0%|          | 0/5 [00:00<?, ?it/s]

Summarize dataset:   0%|          | 0/5 [00:00<?, ?it/s]

Generate report structure:   0%|          | 0/1 [00:00<?, ?it/s]

Render HTML:   0%|          | 0/1 [00:00<?, ?it/s]

Export report to file:   0%|          | 0/1 [00:00<?, ?it/s]

In [116]:
x = df_complete.groupby('booking_complete')[['booking_origin', 'flight_day']].count()
result = df_complete.groupby(['booking_origin', 'flight_day'])['booking_complete'].count().reset_index(name='count')
result 

booking_origin flight_day  count
0        (not set)        Fri      3
1        (not set)        Sat      1
2        (not set)        Sun      2
3        (not set)        Thu      1
4        (not set)        Tue      1
..             ...        ...    ...
225        Vietnam        Sat     18
226        Vietnam        Sun      7
227        Vietnam        Thu     16
228        Vietnam        Tue     18
229        Vietnam        Wed     17

[230 rows x 3 columns]

In [130]:
df.columns

Index(['num_passengers', 'sales_channel', 'trip_type', 'purchase_lead',
       'length_of_stay', 'flight_hour', 'flight_day', 'departure',
       'destination', 'booking_origin', 'wants_extra_baggage',
       'wants_preferred_seat', 'wants_in_flight_meals', 'flight_duration',
       'booking_complete'],
      dtype='object')

# Rapport

Ce dataset semble concerner des réservations de vols et contient des informations détaillées sur les passagers, les réservations effectuées, les préférences des passagers, ainsi que les caractéristiques des vols eux-mêmes.

### Objectif
L'objectif de ce dataset est d'analyser les facteurs déterminants pour savoir si une réservation sera complète ou non. En explorant ces variables, nous cherchons à identifier les éléments qui influencent la finalisation d'une réservation, comme les préférences des passagers, le canal de vente, ou encore le délai d'achat avant le départ.

# AED

### Analyse du Dataset
- **Dimensions du Dataset** : Le dataframe comporte 50 000 lignes et 14 colonnes.
- **Données Manquantes** : Aucune donnée manquante n’a été trouvée dans le dataset.
- **Lignes Dupliquées** : Il y avait 719 lignes dupliquées, qui ont été supprimées.

### Durée des Vols :
- **Durée maximale du vol** : 9.5 heures.
- **Durée minimale du vol** : 4.67 heures.

### Durée du Séjour :
- **Durée maximale du séjour** : 778 jours.
- **Durée minimale du séjour** : 0 jour.

### Pays Différents pour `booking_origin` :
Le dataset contient des réservations provenant de 66 pays différents.

### Résultats des Réservations par Pays :

#### Pays avec le plus de réservations complètes :
- **Malaysia** : 2 434 réservations complètes.
- **Australia** : 895 réservations complètes.
- **China** : 684 réservations complètes.

#### Pays avec le plus de réservations incomplètes :
- **Australia** : 16 796 réservations incomplètes.
- **Malaysia** : 4 621 réservations incomplètes.
- **South Korea** : 4 043 réservations incomplètes.

### Nombre de Passagers par Réservation

#### Réservations Incomplètes (par nombre de passagers) :
- 1 passager : 26 476 réservations (63.2%).
- 2 passagers : 10 640 réservations (25.4%).
- 3 passagers : 2 396 réservations (5.7%).
- 4 passagers : 1 494 réservations (3.6%).
- 5 passagers : 438 réservations (1.0%).
- 6 passagers : 228 réservations (0.5%).
- 7 passagers : 91 réservations (0.2%).
- 8 passagers : 76 réservations (0.2%).
- 9 passagers : 51 réservations (0.1%).

#### Réservations Complètes (par nombre de passagers) :
- 1 passager : 4 403 réservations (59.6%).
- 2 passagers : 2 029 réservations (27.5%).
- 3 passagers : 486 réservations (6.6%).
- 4 passagers : 273 réservations (3.7%).
- 5 passagers : 106 réservations (1.4%).
- 6 passagers : 53 réservations (0.7%).
- 7 passagers : 16 réservations (0.2%).
- 9 passagers : 13 réservations (0.2%).
- 8 passagers : 12 réservations (0.2%).

## Réservations par Canal de Réservation

#### Réservations Incomplètes (par canal de réservation) :
- **Internet** : 37 115 réservations (88.6%).
- **Mobile** : 4 775 réservations (11.4%).

#### Réservations Complètes (par canal de réservation) :
- **Internet** : 6 802 réservations (92.0%).
- **Mobile** : 589 réservations (8.0%).

## Réservations par Type de Voyage

#### Réservations Incomplètes (par type de voyage) :
- **RoundTrip** : 41 413 réservations (98.9%).
- **OneWay** : 366 réservations (0.9%).
- **CircleTrip** : 111 réservations (0.3%).

#### Réservations Complètes (par type de voyage) :
- **RoundTrip** : 7 366 réservations (99.7%).
- **OneWay** : 20 réservations (0.3%).
- **CircleTrip** : 5 réservations (0.1%).

## Réservations par Jour de Vol

#### Réservations Incomplètes (par jour de vol) :
- **Lundi (mon)** : 6 794 réservations (16.2%).
- **Mardi (tue)** : 6 442 réservations (15.4%).
- **Mercredi (wed)** : 6 332 réservations (15.1%).
- **Jeudi (thu)** : 6 214 réservations (14.8%).
- **Vendredi (fri)** : 5 709 réservations (13.6%).
- **Dimanche (sun)** : 5 531 réservations (13.2%).
- **Samedi (sat)** : 4 868 réservations (11.6%).

#### Réservations Complètes (par jour de vol) :
- **Mercredi (wed)** : 1 230 réservations (16.6%).
- **Lundi (mon)** : 1 194 réservations (16.2%).
- **Mardi (tue)** : 1 116 réservations (15.1%).
- **Jeudi (thu)** : 1 109 réservations (15.0%).
- **Vendredi (fri)** : 976 réservations (13.2%).
- **Dimanche (sun)** : 911 réservations (12.3%).
- **Samedi (sat)** : 855 réservations (11.6%).

## Réservations par Pays d'Origine

#### Réservations Incomplètes (par pays d'origine) :
- **Australia** : 16 796 réservations (35.0%).
- **Malaysia** : 4 621 réservations (9.6%).
- **South Korea** : 4 046 réservations (8.4%).
- **Japan** : 3 347 réservations (7.0%).
- **China** : 2 600 réservations (5.4%).
- **Taiwan** : 1 822 réservations (3.8%).
- **Indonesia** : 1 708 réservations (3.6%).
- **Thailand** : 1 529 réservations (3.2%).
- **India** : 1 130 réservations (2.4%).

#### Réservations Complètes (par pays d'origine) :
- **Malaysia** : 2 434 réservations (30.0%).
- **Australia** : 895 réservations (11.0%).
- **China** : 684 réservations (8.4%).
- **Indonesia** : 609 réservations (7.5%).
- **Japan** : 472 réservations (5.8%).
- **Thailand** : 464 réservations (5.7%).
- **South Korea** : 459 réservations (5.6%).
- **Singapore** : 295 réservations (3.6%).
- **Taiwan** : 220 réservations (2.7%).

## Durée des Vols (Flight Duration)

#### Durée des Vols pour les Réservations Incomplètes :
- **8.83 heures** : 12 668 réservations (30.2%).
- **8.58 heures** : 6 354 réservations (15.2%).
- **5.62 heures** : 4 950 réservations (11.8%).
- **6.62 heures** : 3 859 réservations (9.2%).
- **7 heures** : 2 587 réservations (6.2%).
- **7.57 heures** : 2 287 réservations (5.5%).
- **4.67 heures** : 2 032 réservations (4.9%).
- **6.42 heures** : 1 339 réservations (3.2%).
- **5.33 heures** : 1 125 réservations (2.7%).
- **6.33 heures** : 984 réservations (2.3%).

#### Durée des Vols pour les Réservations Complètes :
- **8.83 heures** : 1 671 réservations (22.6%).
- **6.62 heures** : 799 réservations (10.8%).
- **7 heures** : 727 réservations (9.8%).
- **4.67 heures** : 667 réservations (9.0%).
- **8.58 heures** : 538 réservations (7.3%).
- **5.62 heures** : 514 réservations (7.0%).
- **7.57 heures** : 498 réservations (6.7%).
- **4.75 heures** : 416 réservations (5.6%).
- **5.33 heures** : 381 réservations (5.2%).
- **6.42 heures** : 369 réservations (5.0%).


# Structure du Dataset :

Le dataset comporte 50 000 lignes et 14 colonnes.
Il n'y a aucune donnée manquante, mais il y a 719 lignes dupliquées qui ont été supprimées.
Durée du Vol (Flight Duration) :

La durée du vol la plus fréquente pour les réservations incomplètes est 8.83 heures (30.2%), suivie de 8.58 heures (15.2%).
Pour les réservations complètes, 8.83 heures est également la durée la plus fréquente (22.6%), suivie de 6.62 heures (10.8%).
Longueur du Séjour (Length of Stay) :

La longueur du séjour maximale est de 778 jours et la minimale est de 0 jour. Ces extrêmes suggèrent une variabilité importante dans la durée des séjours.
Nombre de Passagers (Num Passengers) :

Pour les réservations incomplètes, la majorité des réservations ont 1 passager (63.2%), suivi de 2 passagers (25.4%).
Pour les réservations complètes, 1 passager (59.6%) est également le plus fréquent, mais avec une proportion plus élevée de 2 passagers (27.5%).
Canal de Réservation (Sales Channel) :

Internet est le canal dominant pour les deux types de réservations (complètes et incomplètes), représentant 88.6% des réservations incomplètes et 92% des réservations complètes.
Le mobile est beaucoup moins fréquent, représentant 11.4% des réservations incomplètes et 8% des complètes.
Type de Voyage (Trip Type) :

Le RoundTrip (voyage aller-retour) est de loin le type de voyage le plus courant pour les deux types de réservation, représentant 98.9% des réservations incomplètes et 99.7% des complètes.
Les OneWay (allers simples) et CircleTrip (voyages circulaires) sont beaucoup moins fréquents.
Jour du Vol (Flight Day) :

Pour les réservations incomplètes, le jour du vol le plus fréquent est lundi (16.2%), suivi de mardi (15.4%) et mercredi (15.1%).
Pour les réservations complètes, mercredi (16.6%) est légèrement plus fréquent que les autres jours, suivi de lundi (16.2%) et mardi (15.1%).
Origine de la Réservation (Booking Origin) :

Le pays ayant le plus de réservations incomplètes est Australie (35.0%), suivi de Malaisie (9.6%) et Corée du Sud (8.4%).
Les réservations complètes proviennent principalement de Malaisie (30.0%) et Australie (11.0%).
Durée du Vol (Flight Duration) :

Les durées de vol les plus fréquentes pour les réservations incomplètes sont 8.83 heures (30.2%) et 8.58 heures (15.2%).
Pour les réservations complètes, la durée la plus fréquente est 8.83 heures (22.6%), suivie de 6.62 heures (10.8%).
Facteurs Influents pour la Complétude des Réservations :

Les réservations incomplètes semblent être liées à des durées de vol plus longues (comme 8.83 heures et 8.58 heures), des séjours plus longs, et une majorité de réservations par internet.
Les réservations complètes, bien que similaires dans la distribution des durées de vol, ont une proportion plus faible de passagers et tendent à être plus concentrées sur Internet.